fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/BaseTableLoader.java

```diff
@@ -74,27 +73,35 @@ protected BaseTableLoader(DateTimeZone defaultTimeZone, DataSourceMetadataServic
     /**
      * Builds a table group.
      * <p>
-     * Builds and loads the physical tables for the physical table definitions as well.
+     * Builds and loads the physical tables from table definitions for the current table group.
      *
-     * @param apiMetrics  The set of metric names surfaced to the api
-     * @param druidMetrics  Names of druid datasource metric columns
-     * @param tableDefinitions  A list of config objects for physical tables
+     * @param tableNames  Set of table name of tables belonging to this table group
+     * @param tableDefinitions  A list of config objects for building physical tables and its dependent physical tables
      * @param dictionaries  The container for all the data dictionaries
+     * @param apiMetrics  The set of metric names surfaced to the api
      *
      * @return A table group binding all the tables for a logical table view together.
      */
     public TableGroup buildDimensionSpanningTableGroup(
-            Set<ApiMetricName> apiMetrics,
-            Set<FieldName> druidMetrics,
+            Set<TableName> tableNames,
```

Comment:
I see. Perhaps we could rename `tableNames` to something like `tablesInTableGroup` or something that indicates their purpose more clearly?